src/base/multithreading.cc

```diff
@@ -310,7 +382,11 @@ MutexLock::MutexLock(const ID i) : id_(i) {
 					}
 
 					s_mutex_.unlock();
-					log_err("%s", info.c_str());
+					if (id_ != ID::kLog) {
+						log_err("%s", info.c_str());
```

Comment:
Maybe this should be always `std::cout`? We are about to throw an exception for a deadlock, that's more important than not messing up the output if that risks a lesser deadlock.